### PR TITLE
Refs #24121 -- Added__repr__() to Lookup

### PR DESCRIPTION
**Pull request by saeedblanchette**

### CODE REVIEW

File: django/db/models/lookups.py [hunk 1]
```diff
@@ -40,6 +40,9 @@ def apply_bilateral_transforms(self, value):
             value = transform(value)
         return value
 
+    def __repr__(self):
+        return f'{self.__class__.__name__}(lhs={self.lhs!r}, rhs={self.rhs!r})'
```
Comment (charettes) [inline django/db/models/lookups.py:44]: Not convinced kwargs notation adds any value to the representation, I'd personally drop them

```suggestion
        return f'{self.__class__.__name__}({self.lhs!r}, {self.rhs!r})'
```

**Comment by felixxm** [comment]
@saeedblanchette Thanks for this patch, however `Lookup.__repr__()` is already handled in #14111.

**Comment by saeedblanchette** [comment]
@felixxm  you're most welcome. I think the [ticket](https://code.djangoproject.com/ticket/24121) should be updated. to avoid the duplicate effort

**Comment by felixxm** [comment]
@saeedblanchette After reconsidering, I decided that your patch is preferable for the `Lookup()` class because it will allow to create an object with the same value from `__repr__()`. Reopened in #14439.

**Pull request closed without merging**